safetensors/src/lib.rs

```diff
@@ -171,7 +173,7 @@ impl<'data> SafeTensors<'data> {
     /// structure.
     pub fn tensors(&self) -> Vec<(String, TensorView<'_>)> {
         let mut tensors = vec![];
-        for (name, info) in &self.metadata.0 {
+        for (name, info) in &self.metadata.1 {
```

Comment:
Changing for a real struct will make that more readable.

`Metadata(HashMap<String, Tensor>)` is just an easy shortcut to name a "raw" type so that you can differentiate.

For instance you could have

`User(String)`
`Email(String)`

```
fn set_email(email: String)   # Potential bug, `set_email("myuser")` 
vs
fn set_email(email: Email)
```
So that you now have 2 different types which will make your code a tiny bit more readable + type safe (with the added cost of adding `.0` in some places).

Not something worth it most of the time.